src/lib/strings/embed-player.ts

```diff
@@ -516,3 +494,56 @@ export function getGiphyMetaUri(url: URL) {
     }
   }
 }
+
+export function parseTenorGif(urlp: URL):
+  | {success: false}
+  | {
+      success: true
+      playerUri: string
+      dimensions: {height: number; width: number}
+    } {
+  if (urlp.hostname !== 'media.tenor.com') return {success: false}
+  try {
+    let [_, id, filename] = urlp.pathname.split('/')
+
+    if (!id || !filename) {
+      throw new Error('Missing id or filename')
+    }
+
+    if (!id.includes('AAAAC')) {
+      throw new Error('Invalid tenor media type')
+    }
+
+    const h = urlp.searchParams.get('hh')
+    const w = urlp.searchParams.get('ww')
+
+    if (!h || !w) {
+      throw new Error('Missing dimensions')
+    }
+
+    const dimensions = {
+      height: Number(h),
+      width: Number(w),
+    }
+
+    if (isWeb) {
+      if (isSafari) {
+        id = id.replace('AAAAC', 'AAAP1')
+        filename = filename.replace('.gif', '.mp4')
+      } else {
+        id = id.replace('AAAAC', 'AAAP3')
+        filename = filename.replace('.gif', '.webm')
+      }
+    } else {
+      id = id.replace('AAAAC', 'AAAAM')
+    }
+
+    return {
+      success: true,
+      playerUri: `https://t.gifs.bsky.app/${id}/${filename}`,
+      dimensions,
+    }
+  } catch {
+    return {success: false}
```

Comment:
I think this is a slightly risky way to approach this because it would swallow actual coding mistakes and silently start failing validation (in a way that seems like something's broken with the gif). 

Since we're already returning a first-class value representing failure and success, let's replace throws with `return { success: false }` and then remove try/catch (assuming we think our code for parsing is solid).

Is there any downside to that?